crates/router/src/utils/user_role.rs

```diff
@@ -412,10 +412,47 @@ pub async fn fetch_user_roles_by_payload(
         .into_iter()
         .filter_map(|user_role| {
             let (_entity_id, entity_type) = user_role.get_entity_id_and_type()?;
-            (request_entity_type.is_none()
-                || request_entity_type
-                    .is_some_and(|req_entity_type| entity_type == req_entity_type))
-            .then_some(user_role)
+            request_entity_type
+                .map_or(true, |req_entity_type| entity_type == req_entity_type)
+                .then_some(user_role)
         })
         .collect::<HashSet<_>>())
 }
+
+pub fn get_min_entity(
```

Comment:
this can be algo which uses `Ordering` implementation of `EntityType` enum.
Maybe refactor after `Internal` variant is used.